components/remote_settings/src/storage.rs

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use camino::Utf8PathBuf;
+
+use crate::{Attachment, RemoteSettingsRecord, Result};
+
+/// Internal storage type
+///
+/// This will store downloaded records/attachments in a SQLite database.  Nothing is implemented
+/// yet other than the initial API.
+///
+/// Most methods input a `base_url` parameter, which is the current base URL for the API client. If
+/// the `base_url` for a get method does not match the one for a set method, then [Storage] should
+/// pretend like nothing is stored in the database.
+///
+/// The reason for this is the [crate::RemoteSettingsService::update_config] method.  If a consumer
+/// passes a new server or bucket to `update_config`, we don't want to be using cached data from
+/// the previous config.
+///
+/// Notes:
+///   - I'm thinking we'll create a separate SQLite database per collection.  That reduces
+///     contention when multiple clients try to get records at once.
+///   - Still, there might be contention if there are multiple clients for the same collection, or
+///     if RemoteSettingsService::sync() and RemoteSettingsClient::get_records(true) are called at
+///     the same time.  Maybe we should create a single write connection and put it behind a mutex
+///     to avoid the possibility of SQLITE_BUSY.  Or maybe not, the writes seem like they should be
+///     very fast.
+///   - Maybe we should refactor this to use the DAO pattern like suggest does.
+pub struct Storage {}
+
+impl Storage {
+    pub fn new(_path: Utf8PathBuf) -> Result<Self> {
```

Comment:
Adding doc comments here as well?

Reply:
I'm adding comments for the other methods, but I'm not sure what to say here other than "this creates a new storage".  I'll leave it to you to decide the details on what's happening.